packages/wrangler/src/dev/dev.tsx

```diff
@@ -489,22 +603,15 @@ function DevSession(props: DevSessionProps) {
 	]);
 
 	const onBundleStart = useCallback(() => {
+		// NOTE: this callback won't be called if props.experimentalDevEnv
 		devEnv.proxy.onBundleStart({
 			type: "bundleStart",
 			config: startDevWorkerOptions,
 		});
-	}, [devEnv, startDevWorkerOptions]);
+	}, [devEnv, startDevWorkerOptions, props.experimentalDevEnv]);
```

Comment:
@RamIdeas What is the expected behaviour here? I think your comment was incomplete, so I don't understand this. Did you mean to write `if(props.experimentalDevEnv) { devEnv.proxy.onBundleStart(/*...*/) }`

Reply:
> Did you mean to write if(props.experimentalDevEnv) { devEnv.proxy.onBundleStart(/*...*/) }

No, explicitly the other way around. The faking of events is what wrangler main was doing. The experimental flag is newer and is the code path of real events within DevEnv.

Slightly confusing, so I put an if statement around it `if(!props.experimentalDevEnv) { devEnv.proxy.onBundleStart(/*...*/) }` (note the `!`) to be explicit but Samuel and I then then decided it was unnecessary as `onBundleStart` is not called unless `props.experimentalDevEnv === false`. So we settled on a comment instead. The prop left in the dependency array was a mistake but is not an issue since it is static and won't change. I'll make a follow-up PR to remove the extraneous dependency array item.

Reply:
Actually I'm going to put the `if` statement back just to be super explicit and avoid confusion. Even if it's redundant, it being code and not a comment will avoid confusion as to whether it is correct.

Reply:
Addressed here https://github.com/cloudflare/workers-sdk/pull/6116